state/state.go

```diff
@@ -27,3 +33,37 @@ type View interface {
 	NewView(ctx context.Context, changes merkledb.ViewChanges) (merkledb.View, error)
 	GetMerkleRoot(ctx context.Context) (ids.ID, error)
 }
+
+type Key struct {
+	Name string
+	// TODO: consider a dynamically sized []byte
+	Permission Permission
+}
+
+type Permission [PermissionLen]byte
+
+func NewKey(name string, permissions ...int) Key {
+	var key Key
+	key.Name = name
+
+	for _, perm := range permissions {
+		// TODO: handle out of bounds
+		byteIdx := perm % 8
```

Comment:
is supposed to be the same as `PermissionLen`? if so, we should just reference `PermissionLen`